src/lib/graphqlTimeoutMiddleware.ts

```diff
@@ -39,6 +55,11 @@ export const graphqlTimeoutMiddleware = (defaultTimeoutInMS: number) => {
     context,
     info
   ) => {
+    const resolverResult = middlewareResolver(parent, args, context, info)
+    // Only add a timeout if there’s async work in progress.
+    if (!isPendingPromise(resolverResult)) {
+      return resolverResult
```

Comment:
👍 I suppose we should test this change first then set the `RESOLVER_TIMEOUT_MS ` down to 0